packages/valory/skills/staking_abci/behaviours.py

```diff
@@ -355,6 +378,22 @@ def _get_service_info(self) -> WaitableConditionType:
         )
         return status
 
+    def _get_agent_ids(self) -> WaitableConditionType:
+        """Get the service information to retrieve the agent ids."""
+        service_id = self.params.on_chain_service_id
+        if service_id is None:
+            self.context.logger.warning(
+                "Cannot perform any staking-related operations without a configured on-chain service id. "
+                "Assuming service status 'UNSTAKED'."
+            )
+            return True
```

Comment:
This logic is now repeated multiple times in the file. Could you introduce a method to avoid duplication?